src/validation/__tests__/ProhibitDeprecatedFieldsRule-test.js

```diff
@@ -0,0 +1,135 @@
+// @flow strict
+
+import { describe, it } from 'mocha';
+
+import { buildSchema } from '../../utilities/buildASTSchema';
+
+import { ProhibitDeprecatedFieldsRule } from '../rules/optional/ProhibitDeprecatedFieldsRule';
+
+import { expectValidationErrorsWithSchema } from './harness';
+
+const schema = buildSchema(`
+  enum EnumType {
+    NORMAL_VALUE
+    DEPRECATED_VALUE @deprecated(reason: "Some enum reason.")
+    DEPRECATED_VALUE_WITH_NO_REASON @deprecated
+  }
+
+  type Query {
+    normalField(enumArg: [EnumType]): String
+    deprecatedField: String @deprecated(reason: "Some field reason.")
+    deprecatedFieldWithNoReason: String @deprecated
+  }
+`);
+
+function expectErrors(queryStr) {
+  return expectValidationErrorsWithSchema(
+    schema,
+    ProhibitDeprecatedFieldsRule,
+    queryStr,
+  );
+}
+
+function expectValid(queryStr) {
+  expectErrors(queryStr).to.deep.equal([]);
+}
+
+describe('Validate: prohibit deprecated fields', () => {
+  it('ignores fields and enum values that are not deprecated', () => {
+    expectValid(`
+      {
+        normalField(enumArg: [NORMAL_VALUE])
+      }
+    `);
+  });
+
+  it('ignores unknown fields and enum values', () => {
+    expectValid(`
+      fragment UnknownFragment on UnknownType {
+        unknownField(unknownArg: UNKNOWN_VALUE)
+      }
+    `);
+  });
+
+  it('reports error when a deprecated field is selected', () => {
+    expectErrors(`
+      {
+        normalField
+        deprecatedField
+      }
+    `).to.deep.equal([
+      {
+        message:
+          'The field Query.deprecatedField is deprecated. Some field reason.',
+        locations: [{ line: 4, column: 9 }],
+      },
+    ]);
+  });
+
+  it('reports error when a deprecated field with default reason is selected', () => {
+    expectErrors(`
+      {
+        normalField
+        deprecatedFieldWithNoReason
```

Comment:
Since this test doesn't actually test anything new (since default for deprecation reason was added during `buildSchema`) it's better to merge it with above test.
It's normal to have two errors in expected result.
Same for enum